src/main/java/dataMgmt/MetadataManager.java

```diff
@@ -156,6 +184,29 @@ public void fileMoved(long fileID, String newFilePath) {
         fileIDtoFilePath.put(fileID, newFilePath);
     }
 
+    public void fileDeleted(long fileID) {
+        String filePath = fileIDtoFilePath.get(fileID);
+        if (filePath != null){
+            fileMetadataMap.remove(filePath);
+        }
+        fileIDtoFilePath.remove(fileID);
```

Comment:
wouldn't this suggestion fail also for `fileMetadataMap.remove(filePath);` if `filePath` is null? seems like they're both wrong. should be
```
String filePath = fileIDtoFilePath.get(fileID);
if (filePath != null){
        fileIDtoFilePath.remove(fileID);
        fileMetadataMap.remove(filePath);
}
```